File: SlSdkDemo/app/src/main/java/com/bmtc/sdk/simple/MainActivity.java
package com.bmtc.sdk.simple;


import android.graphics.Bitmap;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;


import com.bmtc.sdk.contract.PNLShareActivity;
import com.bmtc.sdk.library.SLSDKAgent;
import com.bmtc.sdk.library.base.HttpRequestConfigs;
import com.bmtc.sdk.library.trans.BTContract;
import com.bmtc.sdk.library.trans.IResponse;
import com.bmtc.sdk.library.trans.data.Contract;
import com.bmtc.sdk.library.trans.data.SLUser;
import com.bmtc.sdk.library.uilogic.LogicSDKState;
import com.bmtc.sdk.library.uilogic.LogicWebSocketContract;
import com.bmtc.sdk.library.utils.PreferenceManager;
import com.bmtc.sdk.library.utils.ShareToolUtil;
import com.bmtc.sdk.simple.contract.UsdtActivity;

import java.util.ArrayList;
import java.util.List;

import static com.bmtc.sdk.library.uilogic.LogicSDKState.STATE_LOGIN;


public class MainActivity extends AppCompatActivity implements LogicWebSocketContract.IWebSocketListener, LogicSDKState.ISDKStateListener,View.OnClickListener {
    private TextView tv_usdt_contract,tv_coin_contract,tv_trade_contract;
    private TextView tv_login;
    private TextView tv_share,tv_share2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
        initSLSDKAgent();
    }

    private void initViews() {
        tv_usdt_contract = findViewById(R.id.tv_usdt_contract);
        tv_coin_contract = findViewById(R.id.tv_coin_contract);
        tv_trade_contract = findViewById(R.id.tv_trade_contract);
        tv_login = findViewById(R.id.tv_login);
        tv_share = findViewById(R.id.tv_share);
        tv_share2 = findViewById(R.id.tv_share2);

        tv_usdt_contract.setOnClickListener(this);
        tv_coin_contract.setOnClickListener(this);
        tv_trade_contract.setOnClickListener(this);
        tv_login.setOnClickListener(this);
        tv_share.setOnClickListener(this);
        tv_share2.setOnClickListener(this);
    }

    private void initSLSDKAgent() {
        //设置http接口相关参数
        HttpRequestConfigs httpRequestConfigs = new HttpRequestConfigs()
                .setPrefixHeader("ex")//设置header前缀
                .setHttpReleaseHost("http://co.mybts.info")//设置HTTP接口请求域名
                .setHttpWebSocketHost("ws://ws3.mybts.info/wsswap/realTime")//websocket
                .setExpiredTs("1583901738000000")//过期时间
                .setAccesskey("3e0b5935-6e67-4b55-b345-6f0ed43fafa8");
        httpRequestConfigs.bulid();
        SLSDKAgent.setHttpRequestConfigs(httpRequestConfigs);
        //初始化
        SLSDKAgent.init(this);
        //控制日志输出
        SLSDKAgent.setLogEnabled(true);

        SLSDKAgent.setHttpIsDev(true);
        //构造用户user相关信息
        SLUser user = new SLUser();
     //   user.setToken("7af5683c07c58db9c110149dee090df2");
        String token = "7af5683c07c58db9c110149dee090df2";//PreferenceManager.getString(MainActivity.this,LoginActivity.sTokenKey,"");
        if(!TextUtils.isEmpty(token)){
            user.setToken(token);
            //设置全局user对象
            SLSDKAgent.bindSLUser(user);
        }

        //注册SDK请求客服端事件监听
        SLSDKAgent.registerSLSDKRequestListener(this);
        //注册合约Ticker监听
        SLSDKAgent.registerContractTickerListener(this);
        //获取合约列表
        BTContract.getInstance().contracts(0, new IResponse<List<Contract>>() {
            @Override
            public void onResponse(String errno, String message, List<Contract> data) {
                if (data!=null){
                    List<String> contracts = new ArrayList<>();
                    for (int i = 0 ; i < data.size();i++){
                        contracts.add(data.get(i).getInstrument_id()+"");
                    }
                    if(contracts.size() > 0){
                        //订阅合约ticker
                        SLSDKAgent.sendContractSubscribe(LogicWebSocketContract.WEBSOCKET_TICKER,contracts);
                    }
                }
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        //取消合约监听
        SLSDKAgent.unRegisterContractTickerListener(this);
        //断开WebSocket链接
        SLSDKAgent.disconnectLogicWebSocket();
        SLSDKAgent.unRegisterContractTickerListener(this);
    }

    /**
     * 合约定义信息回调
     * @param data
     */
    @Override
    public void onContractMessage(String data) {

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.tv_usdt_contract:
                UsdtActivity.show(MainActivity.this,0);
                break;
            case R.id.tv_coin_contract:
                UsdtActivity.show(MainActivity.this,1);
                break;
            case R.id.tv_trade_contract:
                UsdtActivity.show(MainActivity.this,2);
                break;
            case R.id.tv_login:
                LoginActivity.show(MainActivity.this);
                break;
            case R.id.tv_share:
                PNLShareActivity.show(MainActivity.this,"",0);
                break;
            case R.id.tv_share2:
                LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
                final View shotView = inflater.inflate(com.bmtc.sdk.contract.R.layout.sl_pnl_share,null);
                final LinearLayout warpLayout = findViewById(R.id.ll_share_placeholder_layout);
                warpLayout.addView(shotView);
                //获取数据 渲染UI 可参考PNLShareActivity类的createItemView方法
                new Handler().postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        View dView = warpLayout;
                        dView.setDrawingCacheEnabled(true);
                        dView.buildDrawingCache();
                        Bitmap bitmap = Bitmap.createBitmap(dView.getDrawingCache());
                        ShareToolUtil.sendLocalShare(MainActivity.this,bitmap);
                    }
                },2000);

                break;
        }
    }


    /**
     * @param actionType LogicSDKState.STATE_LOGIN  登录
     *                   LogicSDKState.STATE_LOGOUT  退出登录
     *                   LogicSDKState.STATE_BIND   绑卡
     */
    @Override
    public void onEvent(int actionType) {
        switch (actionType){
            case STATE_LOGIN:
               LoginActivity.show(MainActivity.this);
                break;

        }
    }
}

File: SlSdkDemo/app/src/main/java/com/bmtc/sdk/simple/MainActivity.kt
package com.bmtc.sdk.simple

import android.content.Intent
import android.graphics.Bitmap
import android.os.Bundle
import android.os.Handler
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import com.bmtc.sdk.contract.PNLShareActivity
import com.bmtc.sdk.contract.UserContractActivity
import com.bmtc.sdk.contract.utils.ShareToolUtil
import com.bmtc.sdk.contract.utils.ToastUtil
import com.bmtc.sdk.simple.contract.UsdtActivity
import com.contract.sdk.ContractPublicDataAgent
import com.contract.sdk.ContractSDKAgent
import com.contract.sdk.data.ContractUser
import com.contract.sdk.impl.ContractSDKListener
import com.contract.sdk.impl.ContractUserStatusListener
import com.contract.sdk.net.ContractHttpConfig
import com.contract.sdk.utils.SDKLogUtil
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(),View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViews()
        initSLSDKAgent()
    }

    private fun initViews() {
        tv_usdt_contract.setOnClickListener(this)
        tv_coin_contract.setOnClickListener(this)
        tv_trade_contract.setOnClickListener(this)
        tv_contract_account.setOnClickListener(this)
        //tv_login.setOnClickListener(this)
        tv_share.setOnClickListener(this)
        tv_share2.setOnClickListener(this)
        tv_simulation_login.setOnClickListener(this)
    }

    /**
     * 初始化SDK 可在Application中进行初始化
     */
    private fun initSLSDKAgent() {
        val contractHttpConfig = ContractHttpConfig()
        contractHttpConfig.prefixHeader = "ex"
        contractHttpConfig.contractUrl = "http://swapapi.nmghhzy.cn/"//设置HTTP接口请求域名
        contractHttpConfig.contractWsUrl = "wss://swapws.hongyachina.com.cn/realTime" //设置合约ws
        //合约SDK Http配置初始化
        ContractSDKAgent.httpConfig = contractHttpConfig
        //是否是合约云SDK
        ContractSDKAgent.isContractCloudSDK = true
        //通知合约SDK语言环境 目前只支持中英文
        ContractSDKAgent.isZhEnv = true
        //合约SDK 必须设置 在最后调用
        ContractSDKAgent.init(this.application)
        //控制日志输出
        ContractSDKAgent.logEnabled = true


        /**
         * 注册监听
         */
        ContractSDKAgent.sdkListener = object : ContractSDKListener{
            override fun sdkInitFail(message: String) {
               ToastUtil.shortToast(this@MainActivity, "SDK初始化失败:$message")
            }

            /**
             * SDK初始化成功后，可获取合约市场Ticker等相关数据
             */
            override fun sdkInitSuccess() {
                ToastUtil.shortToast(this@MainActivity, "SDK初始化成功")

                //订阅所有Ticker
                ContractPublicDataAgent.subscribeAllTickerWebSocket()
            }
        }

        /**
         * 监听登录
         */
        ContractSDKAgent.registerSDKUserStatusListener(this,object:ContractUserStatusListener(){
            /**
             * 合约SDK登录成功
             */
            override fun onContractLoginSuccess() {
                tv_login_status.text = "已登录"
            }

            /**
             * 合约SDK退出登录
             */
            override fun onContractExitLogin() {
                tv_login_status.text = "未登录"
            }

        })
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_usdt_contract -> UsdtActivity.show(this@MainActivity, 0)
            R.id.tv_coin_contract -> UsdtActivity.show(this@MainActivity, 1)
            R.id.tv_trade_contract -> UsdtActivity.show(this@MainActivity, 2)
            R.id.tv_contract_account -> if (ContractSDKAgent.isLogin) {
                val intent = Intent(this@MainActivity, UserContractActivity::class.java)
                startActivity(intent)
            } else {
                ToastUtil.shortToast(this, "需要先登录")
            }
//            R.id.tv_login -> {
//                // LoginActivity.show(MainActivity.this);
//                val userAgent = System.getProperty("http.agent")
//                SDKLogUtil.d("libin", "userAgent:$userAgent")
//            }
            R.id.tv_share -> {
                PNLShareActivity.show(this@MainActivity, "", 0)
            }
            R.id.tv_share2 -> {
                val inflater = LayoutInflater.from(this@MainActivity)
                val shotView = inflater.inflate(com.bmtc.sdk.contract.R.layout.sl_pnl_share, null)
                val warpLayout = findViewById<LinearLayout>(R.id.ll_share_placeholder_layout)
                warpLayout.addView(shotView)
                //获取数据 渲染UI 可参考PNLShareActivity类的createItemView方法
                Handler().postDelayed({
                    val dView: View = warpLayout
                    dView.isDrawingCacheEnabled = true
                    dView.buildDrawingCache()
                    val bitmap = Bitmap.createBitmap(dView.drawingCache)
                    ShareToolUtil.sendLocalShare(this@MainActivity, bitmap)
                }, 2000)
            }
            R.id.tv_simulation_login ->{
                if(!ContractSDKAgent.isLogin){
                    //构造用户user相关信息
                    val user = ContractUser()
                    user.token = "648dee29eca7034f29fc2d94038b32ee"
                    user.expiredTs = "1765075396772000"
                    user.accessKey = "3d2907dd-5ee6-4dfa-b43f-c7180a38ea51"
                    //绑定user对象
                    ContractSDKAgent.user = user
                }

            }
        }
    }

}